File: Selnium script for testing/Script with Python/test_scenario_31.py
from selenium import webdriver
from selenium.webdriver.common.by import By
import time

def execute_test_31():
    """
    Dummy test run for scenario 31
    """
    driver = webdriver.Chrome()
    driver.implicitly_wait(10)
    driver.get("https://dummy.example.com/page_31")

    time.sleep(2)
    driver.find_element(By.LINK_TEXT, "Log Out").click()
    driver.find_element(By.PARTIAL_LINK_TEXT, "Profile").click()
    time.sleep(2)
    driver.find_element(By.ID, "user_name").send_keys("test_data_525")
    driver.find_element(By.PARTIAL_LINK_TEXT, "Profile").click()
    driver.find_element(By.NAME, "password_field").send_keys("test_data_362")
    driver.find_element(By.CLASS_NAME, "nav-item").click()
    time.sleep(2)
    time.sleep(2)
    driver.find_element(By.LINK_TEXT, "Log Out").click()
    driver.find_element(By.ID, "user_name").send_keys("test_data_745")
    driver.find_element(By.ID, "user_name").send_keys("test_data_978")
    assert "Dashboard" in driver.title
    time.sleep(2)

    driver.quit()

if __name__ == "__main__":
    execute_test_31()

File: Selnium script for testing/Script with Java/src/test/java/testscenario/TestScenario31.java
package testscenario;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import java.time.Duration;

public class TestScenario31 {
    public static void executeTest31() {
        /*
         * Dummy test run for scenario 31
         */
        WebDriver driver = new ChromeDriver();
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
        driver.get("https://dummy.example.com/page_31");

        try {
            driver.findElement(By.name("password_field")).sendKeys("test_data_322");
            Thread.sleep(2000);
            if (!driver.findElement(By.cssSelector(".promo-code")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            if (!driver.findElement(By.linkText("Log Out")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            if (!driver.findElement(By.linkText("Log Out")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            driver.findElement(By.id("user_name")).sendKeys("test_data_494");
            driver.findElement(By.className("nav-item")).click();
            driver.findElement(By.id("user_name")).sendKeys("test_data_243");
            if (!driver.findElement(By.linkText("Log Out")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            Thread.sleep(2000);
            driver.findElement(By.name("password_field")).sendKeys("test_data_674");
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            driver.quit();
        }
    }
    
    public static void main(String[] args) {
        executeTest31();
    }
}
